cmd/kubernetes.go

```diff
@@ -0,0 +1,818 @@
+		// if we don't have any, generate them now
+		err := internal.CheckCerts(config.ManagerCertFile, config.ManagerKeyFile)
+		if err != nil {
+			err = internal.GenerateCerts(config.ManagerCAFile, config.ManagerCertFile, config.ManagerKeyFile, config.ManagerCertDomain)
+			if err != nil {
+				die("could not generate certs: %s", err)
+			}
+			info("created a new key and certificate for TLS")
+		}
+
+		// we will spawn wr on the remote server we will create, which means we
+		// need to know the path to ourselves in case we're not in the user's
+		// $PATH
+		exe, err := osext.Executable()
+		if err != nil {
+			die("could not get the path to wr: %s", err)
+		}
+
+		// we then  need to rewrite it to always use the 'wr-linux'
+		// binary, in case we are deploying from a mac.
+		exe = filepath.Dir(exe) + linuxBinaryName
+
+		// get all necessary cloud resources in place
+		mp, err := strconv.Atoi(config.ManagerPort)
+		if err != nil {
+			die("bad manager_port [%s]: %s", config.ManagerPort, err)
+		}
+
+		wp, err := strconv.Atoi(config.ManagerWeb)
+		if err != nil {
+			die("bad manager_web [%s]: %s", config.ManagerWeb, err)
+		}
+
+		// Set up the client and resource files
+		c := kubedeployment.Controller{
+			Client: &client.Kubernetesp{},
+		}
+
+		resourcePath := filepath.Join(config.ManagerDir, "kubernetes_resources")
+		resources := &cloud.Resources{}
+
+		// Authenticate and populate Kubernetesp with clientset and restconfig.
+		c.Clientset, c.Restconfig, err = c.Client.Authenticate(kubeLogger)
+		if err != nil {
+			die("Could not authenticate against the cluster: %s", err)
+		}
+
+		// Check if an existing deployment with the label 'app=wr-manager' exists.
+		// Read in namespace from resource file, if no file exists and no namespace
+		// is passed as a flag create new namespace and redeploy.
+		// If a resource file exists or a namespace is passed as a flag check to see if
+		// there is an existing manager deployment to reconnect to
+
+		var kubeDeploy bool //defaults false
+
+		// If a namespace is passed it takes priority.
+		if len(kubeNamespace) != 0 {
+			_, err = c.Clientset.AppsV1().Deployments(kubeNamespace).Get("wr-manager", metav1.GetOptions{})
+			if err != nil {
+				if errors.IsNotFound(err) {
+					// set the flag that tells the daemon to redeploy.
+					kubeDeploy = true
+				} else {
+					die("wr-manager deployment found but not healthy: %s", err)
+				}
+			}
+		} else {
+			// Look for a set of resources in the manager directory
+			// If found, load them, otherwise use a new empty set.
+			if _, serr := os.Stat(resourcePath); os.IsNotExist(serr) {
+				kubeDeploy = true
+			} else {
+				// Read the namespace resource file
+				file, err := os.Open(resourcePath)
+				if err != nil {
+					die("could not open resource file with path: %s", err)
+				}
+				decoder := gob.NewDecoder(file)
+				err = decoder.Decode(resources)
+				if err != nil {
+					die("error decoding resource file: %s", err)
```

Comment:
It would be helpful to print the path to the resource file in both the die() calls above